app/routes/surveys/components/Question.js

```diff
@@ -0,0 +1,237 @@
+// @flow
+
+import React, { Component } from 'react';
+import Option from './Option';
+import { TextInput, TextArea, CheckBox } from 'app/components/Form';
+import Dropdown from 'app/components/Dropdown';
+import styles from './surveys.css';
+import Icon from 'app/components/Icon';
+import { Link } from 'react-router';
+import { ConfirmModalWithParent } from 'app/components/Modal/ConfirmModal';
+
+type Props = {
+  updateQuestion: Object => void,
```

Comment:
`QuestionEntity => void`?

Reply:
Well I've been unsure about how to handle problems like this for a while. Popping in your suggestion won't work right of the bat, since `QuestionEntity` has a couple of mandatory fields like ID that aren't contructed before the object reaches the back-end. I suppose I could make a second entity like `MakeQuestionEntity`, but that would have to be done for almost all entities and would mostly be the same as the original entities but with all fields optional. Doesn't really seem too useful to me?